guides/dsl.md

```diff
@@ -0,0 +1,241 @@
+  let teacher = find("teacher", {name: "Smith"});
+  return [
+    artemis.remove("teacher").add("teacher", teacher)
+  ];
+});
+```
+
+In summary:
+
+- Add a value: `person.salary += 10` <-> `person.add("salary", 10)`
+- Remove a value: `person.remove("salary, 10)` <-> `person.salary -= 10`
+- Set a value: `person.remove("salary").add("salary", 10)` <-> `person.salary := 10`
+- Remove an attribute `person.remove("salary")` <-> `person.salary := none`
+- Remove a record `person.remove()` <-> `person := none`
+
+## Subblocks
+
+In the DSL `not()`, `union()`, and `choose()` are subblocks, which have their own body. Let's look at each of these.
+
+### not()
+
+The `not()` sub block works similary to `not()` from the Eve syntax; it performs an anti-join on the records inside and outside of the `not()`. To use not, include it in the function list at the beginning of the block.
+
+```javascript
+program.block("Tag students without any citations.", ({find, record, not}) => {
+    let students = find("student");
+    not(() => {
+        find("student", {citation});
+    });
+    return [
+        students.add("tag","good-standing")
+    ];
+})
+```
+
+### choose()
+
+Choose() and union() expressions are behind the mechanics of the if expression in the Eve syntax. In the DSL, we expose these directly. First, `choose()` takes a body of branches, each one comprised of a condition and a return value. The value of the choose corresponds to the first matching branch. 
+
+```javascript
+program.commit("Assign a letter grade.", ({find, choose}) => {
+    let student = find("student");
+    let [grade] = choose(
+        () => { student.grade >= 90; return "A"; },
+        () => { student.grade >= 80; return "B"; },
+        () => { student.grade >= 70; return "C"; },
+        () => { student.grade >= 60; return "D"; },
+        () => "F"
+    )
+    return [
+        student.add("letter-grade", {grade}),
+    ];
+});
+```
+
+### union()
+
+By contrast `union()` takes a body of branches and combines their output into a single variable. One use of union might be to normalize records from different data sources.
+
+```javascript
+program.block("display the student's full names", ({find, record, union}) => {
+  let east = find("student", {school: "West HS"});
+  let west = find("student", {school: "East HS"});
+  let [fullName] = union(
+    () => { east.name; return east.name},
+    () => { west.firstName; return `${{west.firstName}} ${{west.lastName}}`},
+  );
+  return [
+    record("html/element", {tagname: "div", text: fullName}),
+  ];
+});
+```
+
+### Functions
+
+The standard library in Eve has been redone in the DSL. To use library functions, you must now bring in "lib" explicitly when defining your block. From lib you can access the various standard library functions supported by the runtime so far. 
+
+```javascript
+program.block("display the student's full names", ({find, record, lib}) => {
+  find("angle" degrees)
+  let result = math.sin(degrees)
```

Comment:
as imported, this would be `lib.math.sin(...)`. To use it as `math.sin, instead use this destructuring in the parameter list: `{find, record, lib:{math}}`.